scripts/quests/ahtUrhgan/The_Prince_and_the_Hopper.lua

```diff
@@ -65,18 +65,26 @@ quest.sections =
             {
                 function(player, prevZone)
                     if quest:getVar(player, 'Prog') == 1 then
-                        -- This event should only fire when zoning into Wajaoam Woodlands
-                        -- through the secret exit, as the player was instructed.
-                        if prevZone == xi.zone.AHT_URHGAN_WHITEGATE then
+                    local xPos = player:getXPos()
+                    local yPos = player:getYPos()
+                    local zPos = player:getZPos()
+                        if xPos >= 680.0 and yPos >= -19.0 and zPos >= 218.0 and
```

Comment:
Originally I had simplified this because the only non-teleporting prevZone from Whitegate to Wajoam is through that exit; however this doesn't account for Olduum ring (which is the only other way that could occur).  There has to be an easier way though to determine if the player used that exit.

Reply:
im all ears. im not a developer, but knew this code accomplished the task of limiting it to the exit only. if your preference is to have it cleaner with the olduum ring zone as a possibility, i can just remove this.

Reply:
Let me ponder some ideas.  I think there might be a workaround we can work into the quest script that still keeps things simple.  Worst case we define a region using those same bounds.